libsolidity/interface/CompilerStack.cpp

```diff
@@ -240,19 +241,15 @@ bool CompilerStack::parse()
 			}
 		}
 	}
-	if (Error::containsOnlyWarnings(m_errorReporter.errors()))
-	{
-		m_stackState = ParsingSuccessful;
-		return true;
-	}
-	else
-		return false;
+
+	m_stackState = ParsingPerformed;
+	return !setAndReturnErrorFlag();
```

Comment:
```suggestion
	if (Error::containsOnlyWarnings(m_errorReporter.errors()))
		return !m_error;
	else
	{
		m_error = true;
		return false;
	}
```

And get rid of ``setAndReturnErrorFlag`` altogether? (It's only used here anyways)